commit 9019a2a60873f654755aa81bd415d5767f2623e4
Author: Takashi Iwai <tiwai@suse.de>
Date:   Tue Sep 4 17:58:43 2018 +0200

    staging: bcm2835-audio: Propagate parameter setup error
    
    commit fee5638fe552ff8222c3a5bdcc4a34255e248d8c upstream.
    
    When the parameter setup fails, the driver should propagate the error
    code instead of silently ignoring it.
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Tested-by: Stefan Wahren <stefan.wahren@i2se.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/staging/vc04_services/bcm2835-audio/bcm2835-pcm.c b/drivers/staging/vc04_services/bcm2835-audio/bcm2835-pcm.c
index 13c61af4e723..41bcaff6358b 100644
--- a/drivers/staging/vc04_services/bcm2835-audio/bcm2835-pcm.c
+++ b/drivers/staging/vc04_services/bcm2835-audio/bcm2835-pcm.c
@@ -238,7 +238,7 @@ static int snd_bcm2835_pcm_prepare(struct snd_pcm_substream *substream)
 				       runtime->rate,
 				       snd_pcm_format_width(runtime->format));
 	if (err < 0)
-		audio_error(" error setting hw params\n");
+		goto out;
 
 	memset(&alsa_stream->pcm_indirect, 0, sizeof(alsa_stream->pcm_indirect));
 
@@ -255,8 +255,9 @@ static int snd_bcm2835_pcm_prepare(struct snd_pcm_substream *substream)
 		alsa_stream->buffer_size, alsa_stream->period_size,
 		alsa_stream->pos, runtime->frame_bits);
 
+ out:
 	mutex_unlock(&chip->audio_mutex);
-	return 0;
+	return err;
 }
 
 static void snd_bcm2835_pcm_transfer(struct snd_pcm_substream *substream,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1454_9019a2a60873f654755aa81bd415d5767f2623e4.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1454_9019a2a60873f654755aa81bd415d5767f2623e4.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
